### DISCUSSION TITLE
RBF (replace by fee) in dogecoin core doesn't seem to work

### DISCUSSION THREAD

**Discussion by spencerstock** [dogecoin/dogecoin#3373] [Help needed] [answered]
I've been trying to integrate RBF for DOGE into the wallet app I'm a developer for. I have it all working successfully for BTC but it seems like Doge is either using a different spec or hasn't implemented RBF fully yet. Any help appreciated.

**Comment by patricklodder** [comment]
Dogecoin Core currently has the original opt-in RBF implemented, not the "full RBF" Bitcoin is moving to. See for example `qa/rpc-tests/bumpfee.py`.

Could you give an example of what you're trying to achieve exactly? I want to walk through your scenario with you.

**Reply by spencerstock** [reply]
Here's the basic scenario:

User1 generates a transaction
The transaction is marked as "not final" or "eligible for replacement" by setting the nSequence to max-1
The transaction isn't confirmed and is "stuck" due to low miner fees

User1 then generates a "replacement" transaction using the same inputs but allotting more to miner fees
Upon submitting to the doge node we get back an error message about a mempool conflict

**Reply by patricklodder** [reply]
What specific error do you receive?

**Reply by junderw** [reply, marked as answer]
Not sure about dogecoin, but the RBF trigger is max - 2.

max - 1 was reserved for people who want to use locktime but don't want RBF.

**Reply by patricklodder** [reply]
Thank you @junderw - this is the same for Dogecoin.

**Reply by spencerstock** [reply]
Thanks for flagging that. Correcting the nSequence and one other configuration change to ensure sequences were properly packaged in the raw transaction got things sorted out. I'll be moving this change through the release process.

Appreciate the helpful responses. Reach out if there's anything I can do to help the doge community.
